controller/baseController.go

```diff
@@ -0,0 +1,23 @@
+package controller
+
+import (
+	"errors"
+	"net/http"
+)
+
+type CtxKey struct{}
+
+func GetField(r *http.Request, index int) (string, error) {
+
+	fields, ok := r.Context().Value(CtxKey{}).([]string)
+
+	if !ok {
+		return "", errors.New("error: parameter is invalid")
+	}
+
+	if index > len(fields) {
```

Comment:
o ultimo elemento de um slice é len()-1, pois começá em zero. Nesse caso, se index for igual a len(), ele não cai nesse if e toma um panic lá em baixo por índice fora do slice.

Precisa ser index >= len()